src/ring_theory/power_series/basic.lean

```diff
@@ -1010,6 +1012,29 @@ begin
   { intros m hm, rwa nat.eq_zero_of_le_zero (nat.le_of_succ_le_succ hm) }
 end
 
+open finset nat
+
+/-- The ring homomorphism taking a power series `f(X)` to `f(aX)`. -/
+noncomputable def eval_mul_hom (a : R) : power_series R →+* power_series R :=
+{ to_fun :=   λ f, power_series.mk $ λ n, a^n * (power_series.coeff R n f),
+  map_zero' := by { ext, simp only [linear_map.map_zero, power_series.coeff_mk, mul_zero], },
+  map_one' := by { ext1, simp only [mul_boole, power_series.coeff_mk, power_series.coeff_one],
+                split_ifs, { rw [h, pow_zero], }, refl, },
+  map_add' := by {intros, ext, norm_num, rw mul_add, },
+  map_mul' := by {intros, ext, rw [power_series.coeff_mul, power_series.coeff_mk,
+              power_series.coeff_mul, finset.mul_sum], apply sum_congr rfl, norm_num,
+              intros b c H, rw [<-H, pow_add], ring, }, }
+
+lemma eval_mul_hom_zero (f : power_series R) : eval_mul_hom 0 f = C R (constant_coeff  R f) :=
+begin
+  rw [eval_mul_hom, ring_hom.coe_mk], ext, rw [power_series.coeff_mk _ _, coeff_C],
+  split_ifs, { rw h, simp only [one_mul, coeff_zero_eq_constant_coeff, pow_zero], },
+  { rw [zero_pow' n h, zero_mul], },
+end
+
+lemma eval_mul_hom_one (f : power_series R) : eval_mul_hom 1 f = f :=
```

Comment:
As above, the stronger statement is
```
@[simp] lemma eval_mul_hom_one : eval_mul_hom 1 = ring_hom.id (power_series R) :=
```
and can be proved in the same way.

Reply:
Done!

Reply:
The one I suggest makes the one in this PR provable by `simp`. The are cases where the former makes progress but the latter doesn't without an `ext` first.

Reply:
But having both is fine